src/index.ts

```diff
@@ -60,12 +105,86 @@ async function encryptWithKey<R>(
  * the resulting value
  * @param {string} password - password to decrypt with
  * @param {string} text - cypher text to decrypt
+ * @param {CryptoKey} key - a key to use for decrypting
+ * @returns {object}
  */
-async function decrypt<R>(password: string, text: string): Promise<R> {
+async function decrypt(
+  password: string,
+  text: string,
+  key?: CryptoKey,
+): Promise<unknown> {
+  const payload = JSON.parse(text);
+  const { salt } = payload;
+
+  const cryptoKey = key || (await keyFromPassword(password, salt));
+
+  const result = await decryptWithKey(cryptoKey, payload);
+  return result;
+}
+
+/**
+ * Given a password and a cypher text, decrypts the text and returns
+ * the resulting value, keyString, and salt
+ * @param {string} password - password to decrypt with
+ * @param {string} text - cypher text to decrypt
+ * @returns {object}
+ */
+async function decryptWithDetail(
+  password: string,
+  text: string,
+): Promise<DetailedDecryptResult> {
   const payload = JSON.parse(text);
   const { salt } = payload;
   const key = await keyFromPassword(password, salt);
-  return await decryptWithKey(key, payload);
+  const extractedKeyString = await exportKey(key);
+  const vault = await decrypt(password, text, key);
+
+  return {
+    extractedKeyString,
+    vault,
+    salt,
+  };
+}
+
+/**
+ * Receives an exported CryptoKey string, creates a key,
+ * and decrypts cipher text with the reconstructed key
+ * @param {string} password - password to decrypt with
+ * @param {string} text - cypher text to decrypt
+ * @returns {object}
+ */
+async function decryptWithEncryptedKeyString(keyString: string, data: string) {
```

Comment:
Is this function needed? It seems like it just wraps two other public function calls. Given that `createKeyFromString` is already called in KeyringController for encryption, using for decryption might make this API easier to understand, making the two operations more symmetric and leaving one less method to understand.

Reply:
Good point!  Updated in both PRs.